### PR TITLE
[RLlib] Fix `compute_actions()` for Trainer due to missing `if prev_actions/rewards is not None` checks.

### PR DESCRIPTION
**Pull request by n30111**
## Why are these changes needed?

Trainer `compute_actions` method fails when user pass `numpy array` as inputs for `prev_action` and `prev_reward`  arguments. It also fails if `List` is passed as input `ValueError: The two structures don't have the same nested structure.` 

### Script to reproduce
```
import gym
import numpy as np
from ray.rllib.agents import ppo

config = ppo.DEFAULT_CONFIG.copy()
trainer = ppo.PPOTrainer(config, env = "LunarLanderContinuous-v2")
for i in range(1):
  trainer.train()

env = gym.make("LunarLanderContinuous-v2")

obs = {"0": env.reset()}
prev_action = np.asarray([[0.1, 0.2]])

# This work
results = trainer.compute_actions(obs, None, prev_action=None)
print(results)

# This fails
results = trainer.compute_actions(obs, None, prev_action=prev_action)
print(results)
```

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by gjoliver** [comment]
just a question about your PR description.
at which point do these numpy input arrays become None?

**Comment by n30111** [comment]
> just a question about your PR description.
> at which point do these numpy input arrays become None?

So, what I means is that, If I pass a numpy array as input to `compute_actions` (prev_action argument), it will throw an error `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
```
agent = PPOTrainer(...)
agent.compute_actions(obs, states, prev_action=np_array)
```

**Comment by gjoliver** [comment]
> > just a question about your PR description.
> > at which point do these numpy input arrays become None?
> 
> So, what I means is that, If I pass a numpy array as input to `compute_actions` (prev_action argument), it will throw an error `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
> 
> ```
> agent = PPOTrainer(...)
> agent.compute_actions(obs, states, prev_action=np_array)
> ```

Sure. Can you comment above the if statements why checking for None values would help in this case?
Thanks.

**Comment by n30111** [comment]
> Sure. Can you comment above the if statements why checking for None values would help in this case? Thanks.

Python can not interpret  numpy array as a whole for true/false evaluations, the check has to be performed element wise. Since in the above method needs numpy array as argument and there is a None check, so we either need to replace with generic elements None check (`if x is  not None`) or Numpy specific `if prev_action.all():` check.

**Comment by n30111** [comment]
I have added a script in the PR description to reproduce the error. Thanks.

File: rllib/agents/trainer.py [hunk 1]
```diff
@@ -1732,9 +1732,13 @@ def compute_actions(
             state = [np.stack(s) for s in state]
 
         input_dict = {SampleBatch.OBS: obs_batch}
-        if prev_action:
+
+        # For true/false evaluations of inputs that can be either np.ndarray,
+        # list, check has to be for true check or assuming only array
+        # it has to be prev_action.all()
+        if prev_action is not None:
```
[the remark this answers is not in the document]
Comment (n30111) [reply rllib/agents/trainer.py:1740]: thanks, I have updated the comment.

**Review by gjoliver — APPROVED** [review]
@sven1977 can you help merge please. Thanks.

**Review by sven1977 — APPROVED** [review]
Thanks for the fix @n30111 , looks good. Let's merge :)

**Pull request merged**